lib/gnat/handshake.ex

```diff
@@ -5,8 +5,10 @@ defmodule Gnat.Handshake do
   This module provides a single function which handles all of the variations of establishing a connection to a gnatsd server and just returns {:ok, socket} or {:error, reason}
   """
   def connect(settings) do
-    {:ok, tcp} = :gen_tcp.connect(settings.host, settings.port, settings.tcp_opts)
-    perform_handshake(tcp, settings)
+    case result = :gen_tcp.connect(settings.host, settings.port, settings.tcp_opts, settings.timeout) do
```

Comment:
We shouldn't need the `result =` assignment here is the case condition. It can be assigned on matching below

```elixir
{:ok, tcp} -> perform_handsake(tcp, settings)
result -> result
...
```